src/parser/esprima_cpp/esprima.cpp

```diff
@@ -4505,6 +4504,7 @@ class Parser {
     template <class ASTBuilder>
     ASTNode parseSwitchStatement(ASTBuilder& builder)
     {
+        MetaNode node = this->createNode();
         this->expectKeyword(SwitchKeyword);
```

Comment:
@zherczeg 
Can I ask you one thing about node location for debugger?
In this switch statement, I set its location just before `switch` keyword.
(I also fixed `try` and `throw` statement as well)

Which location would it be proper for node, just before the keyword or after it?

Reply:
I think it is up to you to decide, there is no standard for debuggers. As far as I remember, usually debuggers stop at expressions, not statements. E.g:
```
if (
 aa) {}
```
we stop at `aa` not `if`. But you can stop at `if` if you prefer that.

Reply:
@zherczeg 
Thanks for your good advice :)
I fixed this patch to make debugger stop at expressions!